src/variableinspector.ts

```diff
@@ -80,6 +87,16 @@ export class VariableInspectorPanel
     this.intializeFilteredTable();
   }
 
+  set settingsPromise(
+    settingPromise: PromiseDelegate<ISettingRegistry.ISettings>
+  ) {
+    this._settingsPromise = settingPromise;
+    this._settingsPromise.promise.then(settings => {
+      this._settings = settings;
+      this.initalizeFilterButtons();
+    });
+  }
+
```

Comment:
This is not a great pattern as we should avoid making this class aware of the settings registry.

To do this, this class need to have public setter for the filters and a signal that is emitted when they change. This signal will be used to store new filters to the settings.

The pseudo code would look like that:

```ts
// index.ts

panel.filters = settings.composite['filters];
panel.filtersChanged.connect((_, newFilters) => { settings.set('filters', newFilters); });

// variableinspector.ts

class VariableInspectorPanel {
  private _filtersChanged = new Signal<this, string[]>(this);
  private _filters = new Array<string>();

  set filters(value: string[]) {
    this._filters = value;
    this._filtersChanged.emit(this._filters);
  }

  get filterChanged(): ISignal<this, string[]> {
    return this._filtersChanged;
  }
}
```
}

Reply:
@fcollonval Sorry for the delay. I was just wondering how we are able to access the settings in the index for the two lines.

```
panel.filters = settings.composite['filters];
panel.filtersChanged.connect((_, newFilters) => { settings.set('filters', newFilters); });
```

Currently, the settings are a promise in which we load the settings.

```
const settings = new PromiseDelegate<ISettingRegistry.ISettings>();

Promise.all([settingRegistry.load(variableinspector.id), app.restored])
      .then(([_settings]) => {
        return settings.resolve(_settings);
      })
      .catch(error => {
        console.error('Failed to load settings for the Git Extension', error);
        settings.reject(error);
      });
```

However, when we try to access the settings with the filter, the promise has not resolved yet. How do we go about this?

Reply:
Hey @kentarolim10 - nice to read you.
---
A common pattern is to call a `onSettingsChanged` method in the plugin activation function that propagate the settings to the appropriate objects when they changed; see e.g. https://github.com/jupyterlab-contrib/search-replace/blob/8f4bdff704339cb189e4d54beb230a1e510000f7/src/index.ts#L75

In this case to set the settings on all variable-inspector panels, you could make use of the [`tracker.forEach`](https://jupyterlab.readthedocs.io/en/latest/api/classes/apputils.WidgetTracker-1.html#forEach) method.